core/capabilities/remote/target/client_test.go

```diff
@@ -190,6 +186,7 @@ func testClient(ctx context.Context, t *testing.T, numWorkflowPeers int, workflo
 	// Fire off all the requests
 	for _, caller := range callers {
 		go func(caller commoncap.TargetCapability) {
+			defer wg.Done()
```

Comment:
By deferring the `Done` call, we ensure that a messy shutdown (like test suite timeout) will still unblock `Wait`.